packages/web/src/lib/myalgowallet-mode.ts

```diff
@@ -0,0 +1,135 @@
+    const groupID = algosdk.computeGroupID(txnsGroup);
+    for (let i = 0; i < txns.length; i++) txnsGroup[i].group = groupID;
+    return await this.connector.signTransaction(txnsGroup.map((txn) => txn.toByte()));
+  }
+
+  /**
+   * Send signed transaction to network and wait for confirmation
+   * @param rawTxns Signed Transaction(s)
+   */
+  private async sendAndWait (
+    rawTxns: Uint8Array | Uint8Array[]
+  ): Promise<algosdk.modelsv2.PendingTransactionResponse> {
+    const txInfo = await this.algodClient.sendRawTransaction(rawTxns).do();
+    return await this.waitForConfirmation(txInfo.txId);
+  }
+
+  // Function used to wait for a tx confirmation
+  private async waitForConfirmation (
+    txId: string
+  ): Promise<algosdk.modelsv2.PendingTransactionResponse> {
+    const response = await this.algodClient.status().do();
+    let lastround = response[LAST_ROUND];
+    while (true) {
+      const pendingInfo = await this.algodClient.pendingTransactionInformation(txId).do();
+      if (pendingInfo["pool-error"]) {
+        throw new Error(`Transaction Pool Error: ${pendingInfo["pool-error"] as string}`);
+      }
+      if (pendingInfo[CONFIRMED_ROUND] !== null && pendingInfo[CONFIRMED_ROUND] > 0) {
+        return pendingInfo as algosdk.modelsv2.PendingTransactionResponse;
+      }
+      lastround++;
+      await this.algodClient.statusAfterBlock(lastround).do();
+    }
+  }
+
+  /**
+   * Execute single transaction or group of transactions (atomic transaction)
+   * @param execParams transaction parameters or atomic transaction parameters
+   */
+  async executeTransaction (
+    execParams: ExecParams | ExecParams[]
+  ): Promise<algosdk.modelsv2.PendingTransactionResponse> {
+    let signedTxn;
+    let txns: Transaction[] = [];
+    let confirmedTx: algosdk.modelsv2.PendingTransactionResponse;
+    if (Array.isArray(execParams)) {
+      if (execParams.length > 16) {
+        throw new Error("Maximum size of an atomic transfer group is 16");
+      }
+      for (const [_, txn] of execParams.entries()) {
+        txns.push(mkTransaction(txn, await mkTxParams(this.algodClient, txn.payFlags)));
+      }
+
+      txns = algosdk.assignGroupID(txns);
+      const toBeSignedTxns: TransactionInGroup[] = txns.map((txn: Transaction) => {
+        return { txn: txn, shouldSign: true };
+      });
+
+      signedTxn = await this.signTransactionGroup(toBeSignedTxns);
+
+      signedTxn = signedTxn.filter((stxn) => stxn);
+      const Uint8ArraySignedTx = signedTxn.map((stxn) => stxn.blob);
+      confirmedTx = await this.sendAndWait(Uint8ArraySignedTx);
+    } else {
+      const txn = mkTransaction(
+        execParams,
+        await mkTxParams(this.algodClient, execParams.payFlags)
+      );
+      signedTxn = await this.signTransaction(txn);
+      const Uint8ArraySignedTx = signedTxn.blob;
+      confirmedTx = await this.sendAndWait(Uint8ArraySignedTx);
+    }
+
+    console.log("confirmedTx: ", confirmedTx);
```

Comment:
Could we use a logger to handle log instead of `console.log`? How about think about it?

Reply:
I think we can create a new task, where we can update all logs with that logger.

Reply:
Yes, sound goods. Could you do it?